service/server.py

```diff
@@ -12,18 +13,27 @@ class State(Enum):
     UNAVAILABLE = 1
 
 
+class StateLocker:
+    """Synchronizes the state of file_transaction_svc"""
+    lock = rwlock.RWLockWrite()
+    current_service_state = None
+
+    def __init__(self):
+        self.current_service_state = State.AVAILABLE
```

Comment:
You probably do not need a constructor and just assign the state on declaration.